lang/c++/test/DataFileTests.cc

```diff
@@ -217,7 +217,7 @@ class DataFileTest {
 #endif
 
     void testWriteWithCodec(avro::Codec codec) {
-        avro::DataFileWriter<ComplexInteger> df(filename, writerSchema, 100);
+        avro::DataFileWriter<ComplexInteger> df(filename, writerSchema, 100, codec);
```

Comment:
Look at that! A place where an unused parameter warning actually revealed a bug in the tests, not always just noise :)